light/data_model/db/environment.py

```diff
@@ -226,25 +226,35 @@ class DBAgent(DBElem, SQLBase):
     )
     ID_PREFIX = "AGE"
 
-    base_id: str = Column(ForeignKey("agent_names.db_id"), nullable=False)
-    persona = Column(String(PERSONA_LENGTH_CAP), nullable=False, index=True)
-    physical_description = Column(
+    base_id: str = Column(ForeignKey("agent_names.db_id"), nullable=False)  # type: ignore
+    persona: str = Column(String(PERSONA_LENGTH_CAP), nullable=False, index=True)  # type: ignore
+    physical_description: str = Column(  # type: ignore
         String(DESCRIPTION_LENGTH_CAP), nullable=False, index=True
     )
-    name_prefix = Column(String(NAME_PREFIX_LENGTH), nullable=False)
-    is_plural = Column(Boolean)
-    size = Column(Integer)
-    contain_size = Column(Integer)
-    constitution = Column(Float)
-    charisma = Column(Float)
-    strength = Column(Float)
-    dexterity = Column(Float)
-    intelligence = Column(Float)
-    wisdom = Column(Float)
-    base_name: List["DBAgent"] = relationship(
+    name_prefix: str = Column(String(NAME_PREFIX_LENGTH), nullable=False)  # type: ignore
+    is_plural: bool = Column(Boolean)  # type: ignore
+    size: int = Column(Integer)  # type: ignore
+    contain_size: int = Column(Integer)  # type: ignore
+    constitution: float = Column(Float)  # type: ignore
+    charisma: float = Column(Float)  # type: ignore
+    strength: float = Column(Float)  # type: ignore
+    dexterity: float = Column(Float)  # type: ignore
+    intelligence: float = Column(Float)  # type: ignore
+    wisdom: float = Column(Float)  # type: ignore
+    base_name: "DBAgentName" = relationship(
         "DBAgentName", backref="agents", foreign_keys=[base_id]
     )
 
+    def character_features(self, globalFull: bool, full: bool = False) -> str:
+        """Get text feature for a db agent (for model use)"""
+        if globalFull:
+            full = True
+        str = ""
+        str += self.name
+        if full:
+            str += f". {self.persona} "
```

Comment:
nit: why adding the extra space at the end here when there is an rstrip next?

Reply:
Blindly copy-pasting this from the `base_elements.py` file. I can clean this up, thanks for the note